docs/sql/data_types/user.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: docu
+title: User Types
+selected: Documentation/Data Types/User
+expanded: Data Types
+---
+| Name | Description |
+|:---|:---|
+| ENUM | Dictionary Encoding representing all possible string values of a column. |
+
+## Enums
+
+An `ENUM` type represents a dictionary data structure with all possible unique values of a column. For example, a column storing the days of the week can be an Enum holding all possible days. Enums are particularly interesting for string columns with high cardinality. This is because the column only stores a numerical reference to the string in the Enum dictionary, resulting in immense saves in disk storage and faster query performance.
+
+
+### Enum Definition
+Enum types are created using the command: 
+```sql 
+CREATE TYPE ${enum_name} AS ENUM ([${value_1},${value_2},...])
+``` 
+For example:
+```sql
+-- Creates new user defined type 'mood' as an Enum
+CREATE TYPE mood AS ENUM ('sad', 'ok', 'happy');
+
+-- This will fail since the mood type already exists
+CREATE TYPE mood AS ENUM ('sad', 'ok', 'happy', 'anxious');
+
+-- This will fail since Enums can not hold null values
+CREATE TYPE breed AS ENUM ('maltese', NULL);
+
+-- This will fail since Enum values must be unique
+CREATE TYPE breed AS ENUM ('maltese', 'maltese');
+
+```
+
+### Enum Usage
+To use an existing enum, we need to create a table with a column referencing it. For example:
```

Comment:
Maybe simplify this, e.g.:

```
After an enum has been created, it can be used anywhere a standard built-in type is used. For example, we can create a table with a column that references the enum.
```